src/Serialization/containers.jl

```diff
@@ -48,16 +95,10 @@ end
 # Saving and loading vectors
 @register_serialization_type Vector
 
-function load_type_params(s::DeserializerState, T::Type{<: MatVecType})
-  !haskey(s, :params) && return T, nothing
-  subtype, params = load_node(s, :params) do _
-    U = decode_type(s)
-    subtype, params = load_type_params(s, U)
-  end
-  return T{subtype}, params
-end
+# see Array above for load_type_params for general Arrays which include Matrix
+# and Vector
 
-function save_object(s::SerializerState, x::Vector)
+function save_object(s::SerializerState, x::Union{Vector{S}, AbstractArray{S, 1}}) where S
```

Comment:
```suggestion
function save_object(s::SerializerState, x::AbstractVector{S}) where S
```
this is the same, but way more readable IMO